libs/experimental/langchain_experimental/llm_symbolic_math/base.py

```diff
@@ -66,7 +66,52 @@ def _evaluate_expression(self, expression: str) -> str:
                 "Unable to import sympy, please install it with `pip install sympy`."
             ) from e
         try:
-            output = str(sympy.sympify(expression, evaluate=True))
+            allowed_symbols = {
+                # Basic arithmetic and trigonometry
+                "sin": sympy.sin,
+                "cos": sympy.cos,
+                "tan": sympy.tan,
+                "cot": sympy.cot,
+                "sec": sympy.sec,
+                "csc": sympy.csc,
+                "asin": sympy.asin,
+                "acos": sympy.acos,
+                "atan": sympy.atan,
+                # Hyperbolic functions
+                "sinh": sympy.sinh,
+                "cosh": sympy.cosh,
+                "tanh": sympy.tanh,
+                "asinh": sympy.asinh,
+                "acosh": sympy.acosh,
+                "atanh": sympy.atanh,
+                # Exponentials and logarithms
+                "exp": sympy.exp,
+                "log": sympy.log,
+                "ln": sympy.log,  # natural log (alias)
+                "log10": sympy.log,  # log base 10 (use sympy.log)
```

Comment:
both log10 and ln are aliased to sympy.log

Reply:
changed so that log would be a user defined base (default 10), ln will be regular sympy log which defaults to ln. And log 10 to base 10